### PR TITLE
Put `erased` under `-Yerased-terms` flag

### PR DESCRIPTION
**Pull request by nicolasstucki**
`erased` will not be included in Scala 3.0 but could be added in future version.
This feature will only be available with `-Yerased-terms`.

### CODE REVIEW

**Comment by bmeesters** [comment]
Unrelated to the PR itself, but why is erased not planned for Scala 3? I cannot find anything in the SIP meeting minutes.

**Comment by nicolasstucki** [comment]
It never made it to the SIPs. This feature might be considered for a future version of Scala (maybe 3.2+ or later).
It was not included because the main motivation for this feature is an effect system that will also not go in Scala 3.0.

File: tests/pos-custom-args/matchtype.scala
Comment (dwijnand) [inline tests/pos-custom-args/matchtype.scala:1]: Wrong directory.  So this file and the 10 others haven't been tested since 2020 😢

Comment (nicolasstucki) [reply tests/pos-custom-args/matchtype.scala:1]: I will re-enable them

**Pull request merged**